braindecode/preprocessing/eegprep_preprocess.py

```diff
@@ -0,0 +1,1139 @@
+        refdata_max_bad_channels: float | None = 0.075,
+        refdata_max_tolerances: tuple[float, float] | None = (-np.inf, 5.5),
+        num_samples: int = 50,
+        subset_size: float = 0.25,
+        bad_channel_nolocs_threshold: float = 0.45,
+        bad_channel_nolocs_exclude_frac: float = 0.1,
+        max_mem_mb: int = 64,
+    ):
+        can_change_duration = " and ".join(
+            opname
+            for opname in (
+                "resample" if resample_to else "",
+                "bad time window removal" if bad_window_max_bad_channels else "",
+            )
+            if opname
+        )
+        super().__init__(
+            can_change_duration=can_change_duration or None,
+        )
+        self.resample_to = resample_to
+        self.reinterpolate = bad_channel_reinterpolate
+        self.common_avg_ref = common_avg_ref
+        self.burst_removal_cutoff = burst_removal_cutoff
+        self.bad_window_max_bad_channels = bad_window_max_bad_channels
+
+        if bad_channel_corr_threshold is None:
+            line_noise_crit = None
+        else:
+            line_noise_crit = bad_channel_hf_threshold
+        self.clean_artifacts_params = dict(
+            ChannelCriterion=bad_channel_corr_threshold,
+            LineNoiseCriterion=line_noise_crit,
+            BurstCriterion=burst_removal_cutoff,
+            WindowCriterion=bad_window_max_bad_channels,
+            Highpass=highpass_frequencies,
+            ChannelCriterionMaxBadTime=bad_channel_max_broken_time,
+            BurstCriterionRefMaxBadChns=refdata_max_bad_channels,
+            BurstCriterionRefTolerances=refdata_max_tolerances,
+            # BurstRejection='off', # we don't use that one; left at default
+            WindowCriterionTolerances=bad_window_tolerances,
+            FlatlineCriterion=flatline_maxdur,
+            NumSamples=num_samples,
+            SubsetSize=subset_size,
+            NoLocsChannelCriterion=bad_channel_nolocs_threshold,
+            NoLocsChannelCriterionExcluded=bad_channel_nolocs_exclude_frac,
+            MaxMem=max_mem_mb,
+            # Distance='euclidian',
+            # Channels=None,
+            # Channels_ignore=None,
+            # availableRAM_GB=None,
+            # (no other args)
+        )
+
+    def apply_eeg(self, eeg: dict[str, Any], raw: BaseRaw) -> dict[str, Any]:
+        """Apply the preprocessor to an EEGLAB EEG structure. Overridden by subclass."""
+        # remove per-channel DC offset (can be huge)
+        eeg["data"] -= np.median(eeg["data"], axis=1, keepdims=True)
+
+        # optional resampling
+        if (srate := self.resample_to) is not None:
+            eeg = eegprep.resample(eeg, srate)
+
+        # do a check if the data has a supported sampling rate
+        if self.burst_removal_cutoff is not None:
+            from eegprep.utils import round_mat
+
+            supported_rates = (100, 128, 200, 250, 256, 300, 500, 512)
```

Comment:
Can you put the reference on top of this line?

Reply:
Reference for each preprocessing, looks a dummy step, but as the documentation is created based on this, it is better if we have the reference.